src/python/pants/rules/core/test.py

```diff
@@ -3,24 +3,68 @@
 
 import logging
 from dataclasses import dataclass
+from enum import Enum
 from typing import Optional
 
 from pants.base.exiter import PANTS_FAILED_EXIT_CODE, PANTS_SUCCEEDED_EXIT_CODE
 from pants.build_graph.address import Address, BuildFileAddress
 from pants.engine.addressable import BuildFileAddresses
 from pants.engine.build_files import AddressProvenanceMap
 from pants.engine.console import Console
+from pants.engine.fs import Digest
 from pants.engine.goal import Goal, GoalSubsystem
+from pants.engine.isolated_process import FallibleExecuteProcessResult
 from pants.engine.legacy.graph import HydratedTarget
+from pants.engine.objects import union
 from pants.engine.rules import UnionMembership, console_rule, rule
 from pants.engine.selectors import Get, MultiGet
-from pants.rules.core.core_test_model import Status, TestResult, TestTarget
 
 
 # TODO(#6004): use proper Logging singleton, rather than static logger.
 logger = logging.getLogger(__name__)
 
 
+class Status(Enum):
+  SUCCESS = "SUCCESS"
+  FAILURE = "FAILURE"
+
+
+@dataclass(frozen=True)
+class TestResult:
+  status: Status
+  stdout: str
+  stderr: str
+  coverage_digest: Optional[Digest] = None
```

Comment:
Related to the discussion on the design review (#8915): I'd be fine with a digest, and would also be fine with a temporary special field. I also proposed a third option there.

As long as we continue to iterate on the design to ensure that we can allow for unification across coverage engines, I'm ok with incremental steps like this (with clear nearby TODOs pointing back to the design).